File: old_leetcode_solutions/Java/twoSum.java
//https://leetcode.com/problems/two-sum/
//Title: Two Sum
//Difficulty: Easy
//Language: Java
//Author: Pranay Kotian

class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int tempTar;
        int i = 0; 
        int j = 0;
        
        outer: for (i = 0; i < nums.length; i++) {
            tempTar = target - nums[i];
            for (j = i + 1; j < nums.length; j++) {
                if (nums[j] == tempTar)
                    break outer;
            }
        }
        return new int[] {i, j};

        /*
		Runtime: 43 ms, faster than 30.48% of Java online submissions for Two Sum.
		Memory Usage: 39.4 MB, less than 5.65% of Java online submissions for Two Sum.
        */
        
    }
}

File: old_leetcode_solutions/Python/TwoSum.py
#https://leetcode.com/problems/two-sum/
#Title: Two Sum
#Difficulty: Easy
#Language: Python3
#Author: Pranay Kotian

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #BRUTE FORCE SOLUTION (very slow 6400ms)
        # length = len(nums)
        # for i in range(length):
        #     j = i + 1
        #     while j < length:
        #         if target == (nums[i] + nums[j]):
        #             return [i, j];
        #         j += 1
        
        #NOT A HASH TABLE SOLUTION (find better code for duplicate num/compliment case)
        # n = len(nums)
        # for i in range(n):
        #     comp = target - nums[i]
        #     if (comp == nums[i]):
        #         if (comp in nums[i+1:n]):
        #             return [i, nums[i+1:n].index(comp) + i + 1]
        #     elif (comp in nums):
        #         return [i, nums.index(comp)]
        
        #HASH TABLE SOLUTION
        #copy values into hash map
        hashmap = {}
        for i, val in enumerate(nums):
            temp = target - val
            if temp in hashmap:
                return [hashmap[temp], i]
            hashmap[val] = i